tests/integrations/celery/test_celery.py

```diff
@@ -307,9 +309,24 @@ def dummy_task(self):
 
 
 @pytest.mark.forked
-def test_redis_backend_trace_propagation(init_celery, capture_events_forksafe, tmpdir):
+@pytest.mark.parametrize("custom_celery_task_cls", (True, False))
+def test_redis_backend_trace_propagation(
+    init_celery, capture_events_forksafe, tmpdir, custom_celery_task_cls
+):
     celery = init_celery(traces_sample_rate=1.0, backend="redis", debug=True)
 
+    custom_calls = []
+
+    if custom_celery_task_cls:
+        import celery as celery_mod
+
+        class CustomTask(celery_mod.Task):
+            def __call__(self, *args, **kwargs):
+                custom_calls.append(1)
```

Comment:
this is in forked process so this will always be empty. Tried with set_tag and asserted over event payload, could not repro either